examples/oft-solana/tasks/solana/setAuthority.ts

```diff
@@ -0,0 +1,169 @@
+import { AccountMeta, publicKey, transactionBuilder } from '@metaplex-foundation/umi'
+import { fromWeb3JsPublicKey, toWeb3JsKeypair, toWeb3JsPublicKey } from '@metaplex-foundation/umi-web3js-adapters'
+import { AuthorityType, TOKEN_PROGRAM_ID, createSetAuthorityInstruction, getMint } from '@solana/spl-token'
+import { PublicKey } from '@solana/web3.js'
+import bs58 from 'bs58'
+import { task } from 'hardhat/config'
+
+import { types as devtoolsTypes } from '@layerzerolabs/devtools-evm-hardhat'
+import { EndpointId } from '@layerzerolabs/lz-definitions'
+import { OftPDA } from '@layerzerolabs/oft-v2-solana-sdk'
+
+import { checkMultisigSigners, createMintAuthorityMultisig } from './multisig'
+
+import { deriveConnection, getExplorerTxLink } from './index'
+
+const DEFAULT_LOCAL_DECIMALS = 9
+
+interface CreateOFTTaskArgs {
+    /**
+     * The endpoint ID for the Solana network.
+     */
+    eid: EndpointId
+
+    /**
+     * The escrow public key.
+     */
+    escrow: string
+
+    /**
+     * The token mint ID, for Mint-And-Burn-Adapter only.
+     */
+    mint: string
+
+    /**
+     * The program ID for the OFT program.
+     */
+    programId: string
+
+    /**
+     * The CSV list of additional minters.
+     */
+    additionalMinters: string
+
+    /**
+     * The token program ID, for Mint-And-Burn-Adapter only.
+     */
+    tokenProgram: string
+}
+
+/**
+ * Derive the OFT Store account for a given program and escrow.
+ * @param programId {string}
+ * @param escrow {string}
+ */
+const getOftStore = (programId: string, escrow: string) => {
+    const oftDeriver = new OftPDA(publicKey(programId))
+    const escrowPK = publicKey(escrow)
+    const [oftStorePda] = oftDeriver.oftStore(escrowPK)
+    return oftStorePda
+}
+
+/**
+ * Get the string representation of the authority type.
+ * @param authorityType {AuthorityType}
+ */
+const getAuthorityTypeString = (authorityType: AuthorityType) => {
+    switch (authorityType) {
+        case AuthorityType.MintTokens:
+            return 'MintTokens'
+        case AuthorityType.FreezeAccount:
+            return 'FreezeAccount'
+        default:
+            throw Error(`Unknown authority type: ${authorityType}`)
+    }
+}
+
+// Define a Hardhat task for creating and setting a new Mint/Freeze Authority
+// for OFT on Solana
+// * Create SPL Multisig account for mint authority
+// * Sanity check the new Multisig account
+// * Set Mint Authority
+// * Set Freeze Authority
+// Note:  Only supports SPL Token Standard.
+task('lz:oft:solana:setauthority', 'Mints new SPL Token and creates new OFT Store account')
+    .addParam('eid', 'Solana mainnet or testnet', undefined, devtoolsTypes.eid)
+    .addParam('mint', 'The Token mint public key (used for MABA only)')
```

Comment:
this param description seems wrong, it seems mint is required not only for MABA

Reply:
Fixed.